src/SystemCommands-ClassCommands/SycRemoveClassStrategy.class.st

```diff
@@ -0,0 +1,88 @@
+"
+I am the abstract strategy used to remove classes from the system.
+
+My concrete subclasses handle the possible actions when a user attemps to remove a class:
+	- remove the class
+	- remove the class, keeping the subclasses
+	- Don't remove the class, but browse references/subclasses/users
+	- Cancel the action
+"
+Class {
+	#name : #SycRemoveClassStrategy,
+	#superclass : #Object,
+	#category : #'SystemCommands-ClassCommands'
+}
+
+{ #category : #testing }
+SycRemoveClassStrategy class >> canExecuteWithReferences: hasReferences subclasses: hasSubclasses users: hasUsers [
+	self subclassResponsibility 
+]
+
+{ #category : #'instance creation' }
+SycRemoveClassStrategy class >> createForBrowser: aBrowser classes: aCollection [
+
+	| hasReferences hasSubclasses hasUsers |
+	hasReferences := (ClyClassExternalReferencesQuery
+		                  toAny: aCollection
+		                  from: aBrowser systemScope) execute notEmpty.
+	hasSubclasses := ((aCollection flatCollect: [ :each | 
+		                   each subclasses ]) copyWithoutAll: aCollection)
+		                 notEmpty.
+	hasUsers := ((aCollection flatCollect: [ :each | each users ]) 
+		             copyWithoutAll: aCollection) notEmpty.
+
+	hasReferences | hasSubclasses | hasUsers ifFalse: [ 
+		^ { SycSilentlyRemoveClassStrategy new } ].
+
+	^ self
+		  createForBrowser: aBrowser
+		  references: hasReferences
+		  subclasses: hasSubclasses
+		  users: hasUsers
```

Comment:
And then this method can be #createAllForReferences:subclasses:users:browser:. 
(I don't really like so many arguments but I don't have any suggestion to fix that)